FrenetOptimalTrajectory/fot.py

```diff
@@ -111,11 +117,114 @@ def main():
                       np.linalg.norm(initial_conditions['vel']))[0:4]
             )
             plt.grid(True)
-            plt.savefig("{}.jpg".format(i))
+            if save_frame:
+                Path("img/frames").mkdir(parents=True, exist_ok=True)
+                plt.savefig("img/frames/{}.jpg".format(i))
             plt.pause(0.1)
 
     print("Finish")
+
+    print("======================= SUMMARY ========================")
+    print("Total time for {} iterations taken: {}".format(i, total_time))
     print("Average time per iteration: {}".format(total_time / i))
+    print("Max time per iteration: {}".format(max(time_list)))
+
+    return time_list
 
 if __name__ == '__main__':
-    main()
+
+    argument_list = sys.argv[1:]
+    short_options = "dvspct:f"
+    long_options = ["display", "verbose", "save", "profile", "compare", "thread=", "full"]
+
+    enable_verbose, enable_saving, enable_display, enable_compare, enable_profiling, full_profiling = False, False, False, False, False, False
+    thread_num = 0 # default,, run unthreaded planner
+
+    try:
+        arguments, values = getopt.getopt(argument_list, short_options, long_options)
+    except getopt.error as err:
+        print (str(err))
+        sys.exit(2)
+
+    for current_argument, current_value in arguments:
+        if current_argument in ("-d", "--display"):
+            print ("Showing Animation, Ensure you have X11 forwarding server open")
+            enable_display = True
+        elif current_argument in ("-v", "--verbose"):
+            print ("Enabling verbose mode, showing all state info")
+            enable_verbose = True
+        elif current_argument in ("-s", "--save"):
+            print ("Saving each frame of simulation")
+            enable_display, enable_saving = True, True
+        elif current_argument in ("-c", "--compare"):
+            print ("Comparing threaded runtime with unthreaded baseline")
+            enable_compare = True
+        elif current_argument in ("-t", "--thread"):
+            print ("Execute with {} threads".format(current_value))
+            thread_num = int(current_value)
+        elif current_argument in ("-p", "--profile"):
+            print ("Enabling Time Profiling")
+            enable_profiling = True
+        elif current_argument in ("-f", "--full"):
+            print ("Full Time Profiling")
+            full_profiling = True
```

Comment:
This is great, however could you use the [argparse library](https://docs.python.org/3/howto/argparse.html) for the command line arguments? It will make the code more readable and easier to maintain.

Reply:
Thanks for the suggestion! `argparse` is so much cleaner.